src/cdf.rs

```diff
@@ -133,18 +131,39 @@ pub struct CdfModeContext {
 }
 
 #[derive(Clone)]
-#[repr(C)]
-pub struct CdfThreadContext {
-    pub r#ref: *mut Rav1dRef,
-    pub data: CdfThreadContext_data,
-    pub progress: *mut AtomicU32,
+pub enum CdfThreadContext {
+    QCat(c_uint),
+    Cdf(Arc<CdfThreadContext_data>),
+}
+
+impl CdfThreadContext {
+    pub fn progress(&self) -> Option<&AtomicU32> {
+        match self {
+            Self::QCat(_) => None,
+            Self::Cdf(cdf) => cdf.progress.as_ref(),
+        }
+    }
+
+    pub fn cdf_write(&self) -> Option<RwLockWriteGuard<CdfContext>> {
+        match self {
+            Self::QCat(_) => None,
+            Self::Cdf(cdf) => cdf.cdf.try_write().ok(),
+        }
+    }
```

Comment:
If the result is always `.unwrap()`ed, why not put the `.unwrap()` here?  And what about `cdf_read`?

Reply:
cdf read wasn't needed because we only access the field for reading in the same module and we already needed to destructure the enum to do it (`rav1d_cdf_thread_copy`)